src/Tasks/Microsoft.Common.CurrentVersion.targets

```diff
@@ -4149,6 +4156,37 @@ Copyright (C) Microsoft Corporation. All rights reserved.
 
   </Target>
 
+  <!--
+    ============================================================
+                                        _SetPreferNativeArm64Win32ManifestProperties
+
+    Populates Manifest with SupportedArchitectures and produces Win32Manifest property to be used later in the build.
+    ============================================================
+    -->
+
+  <Target
+      Name="_SetPreferNativeArm64Win32ManifestProperties"
+      Condition=" '$(PreferNativeArm64)'=='true' "
+      DependsOnTargets="SetWin32ManifestProperties">
+
+    <PropertyGroup>
+      <SupportedArchitectures>amd64 arm64</SupportedArchitectures>
+    </PropertyGroup>
+
+    <PopulateSupportedArchitectures
+        ApplicationManifestPath="$(Win32Manifest)"
+        OutputDirectory="$(IntermediateOutputPath)"
+        SupportedArchitectures="$(SupportedArchitectures)">
+
+        <Output TaskParameter="ManifestPath" PropertyName="_Win32Manifest"/>
+    </PopulateSupportedArchitectures>
+
+    <PropertyGroup>
+      <Win32Manifest Condition="'$(_Win32Manifest)' != ''">$(_Win32Manifest)</Win32Manifest>
```

Comment:
Mutating the user-supplied property looks non-standard / suspicious. I would probably leave it and use `_Win32Manifest` when invoking the compiler. @rainersigwald, what do you think?

Reply:
Gotcha - actually maybe `Win32Manifest` should be considered internal, despite the missing underscore, and `ApplicationManifest` is what we expect the user to set in their project files.

Looking at this target then, I wonder how this feature interacts with ClickOnce:
https://github.com/dotnet/msbuild/blob/eade7200e9e8c3f7017c7eff60b7cf591b2e71ac/src/Tasks/Microsoft.Common.CurrentVersion.targets#L4133-L4135

Right now it looks like `PreferNativeArm64` has no effect on ClickOnce projects. Ideally I think it should do something or error out.

Reply:
handled in scope of https://github.com/dotnet/msbuild/pull/10134/commits/98cae2341e40d9ab666e25de3166b8ad30945dde

Reply:
Apologies, after some more testing I'm not convinced this is correct. If the app has both an embedded manifest and an external .exe.manifest (the former added by `PreferNativeArm64`, the latter added by ClickOnce targets), Windows will ignore the external one and the app will likely not run correctly.

What we need to do to support this scenario is hook into the targets that add/modify the external manifest to add our content there.

Reply:
The repro is:
1. Create a new Framework Winforms project in VS.
2. Go to Build -> Publish and choose a local path to publish to.
3. Inspect the bin directory of the project - `<app-name>.exe.manifest` is the external Win32 manifest.